File: Question2/src/Student/Student.java
package Student;

public class Student {
    private String name;
    private double cgpa;

    public Student(String name, double cgpa) {
        this.name = name;
        this.cgpa = cgpa;
    }

    public double getCgpa() {
        return cgpa;
    }

    public String getName() {
        return name;
    }

    @Override
    public String toString() {
        return name + " (CGPA: " + cgpa + ")";
    }
}

File: Question3/server/models/Student.js
 import mongoose from "mongoose";

 const studentSchema = new mongoose.Schema({
     StudentName: {
         type: String,
         required: [true, "Name is required"],
         trim: true,
     },
     RegNo: {
         type: String,
         required: true,
         unique: true,
         trim: true,
     },
     semester: {
         type: Number,
         required: true,
         min: 1,
         max: 8,
     },
     cgpa: {
         type: Number,
         required: [true, "CGPA is required"],
         min: 0,
         max: 4.0,
     },
 }, {
     timestamps: true,
 });

 const Student = mongoose.model("Student", studentSchema);
 export default Student;